src/planscape/forsys/parse_forsys_output.py

```diff
@@ -0,0 +1,188 @@
+            'ranked_projects': [],
+            'cumulative_ranked_project_area': [],
+            'cumulative_ranked_project_cost': [],
+        }
+        for i in range(len(self.ranked_projects)):
+            output['ranked_projects'].append(
+                self.ranked_projects[i].to_dictionary())
+            output['cumulative_ranked_project_area'].append(
+                self.cumulative_ranked_project_area[i])
+            output['cumulative_ranked_project_cost'].append(
+                self.cumulative_ranked_project_cost[i])
+        return output
+
+
+class RankedProject():
+    # Project ID.
+    id = -1
+    # Contribution of each priority to the total score
+    # The contribution is weighted according to a scenario's priority weights.
+    weighted_priority_scores = {}
+    # The total score, summed across weighted priority scores.
+    total_score = 0
+    # Project rank.
+    rank = -1
+
+    # Converts a RankedProject to a dictionary.
+    # This facilitates JSON conversion.
+    def to_dictionary(self) -> dict:
+        output = {
+            'id': self.id,
+            'weighted_priority_scores': self.weighted_priority_scores,
+            'total_score': self.total_score,
+            'rank': self.rank,
+        }
+        return output
+
+
+# Transforms the output of a Forsys scenario set run into a more easily-interpreted version.
+class ForsysScenarioSetOutput():
+    scenarios = {}
+
+    # Converts a ForsysScenarioSetOutput to a dictionary.
+    # This facilitates JSON conversion.
+    def to_dictionary(self) -> dict:
+        output = {}
+        for k in self.scenarios.keys():
+            output[k] = self.scenarios[k].to_dictionary()
+        return output
+
+    # Initializes a ForsysScenarioSetOutput instance given raw forsys output, and the following inputs to the forsys call: header names, list of priorities.
+    # Of note, priorities must be listed in the same order they're listed for the forsys call.
+    def __init__(self, raw_forsys_output: "rpy2.robjects.vectors.ListVector", priorities: list[str], project_id_header: str, area_header: str, cost_header: str):
+        self.scenarios = {}
+
+        self.__save_raw_forsys_output_as_dict(raw_forsys_output)
+
+        self.__set_header_names(priorities, area_header,
+                                cost_header, project_id_header)
+
+        for i in range(len(self.__forsys_output_dict[project_id_header])):
+            scenario_weights, scenario_str = self.__get_scenario(i)
+
+            if scenario_str in self.scenarios.keys():
+                self.__append_project_to_existing_scenario(
+                    scenario_str, scenario_weights, i)
+            else:
+                self.__append_project_to_new_scenario(
+                    scenario_str, scenario_weights, i)
+
+    # The raw forsys output consists of 3 R dataframes. This is the index of the "project output" dataframe.
+    __PROJECT_OUTPUT_INDEX = 1
```

Comment:
Do these need to be instance variables?

Reply:
these constants can remain as class variables